seaseq-control.wdl

```diff
@@ -328,45 +329,48 @@ workflow seaseq {
         call fastqc.fastqc as s_mergebamfqc {
             input:
 	        inputfile=s_mergebam.mergebam,
-                default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'\.sorted\.b.*$','') + '/QC/FastQC'
+                default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'.sorted.b.*$','') + '/QC/FastQC'
         }
 
         call samtools.indexstats as s_mergeindexstats {
             input:
                 bamfile=s_mergebam.mergebam,
-                default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'\.sorted\.b.*$','') + '/BAM_files'
+                default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'.sorted.b.*$','') + '/BAM_files'
         }
 
         if ( defined(blacklist) ) {
             # remove blacklist regions
             String string_blacklist = "" #buffer to allow for blacklist optionality
-            File blacklist_ = select_first([blacklist, string_blacklist])
+            File blacklist_m = select_first([blacklist, string_blacklist])
             call bedtools.intersect as s_merge_rmblklist {
                 input :
                     fileA=s_mergebam.mergebam,
-                    fileB=blacklist_,
-                    default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'\.sorted\.b.*$','') + '/BAM_files',
+                    fileB=blacklist_m,
+                    default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'.sorted.b.*$','') + '/BAM_files',
                     nooverlap=true
             }
             call samtools.indexstats as s_merge_bklist {
                 input :
                     bamfile=s_merge_rmblklist.intersect_out,
-                    default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'\.sorted\.b.*$','') + '/BAM_files'
+                    default_location='SAMPLE/' + sub(basename(s_mergebam.mergebam),'.sorted.b.*$','') + '/BAM_files'
             }
         } # end if blacklist provided
 
         File s_mergebam_afterbklist = select_first([s_merge_rmblklist.intersect_out, s_mergebam.mergebam])
 
+        call samtools.index_bam as s_mergebam_afterbklist_index_m { input: bam=s_mergebam_afterbklist }
```

Comment:
this is redundant. 
ROSE = select_first([s_merge_bklist.indexbam, s_mergeindexstats.indexbam])
BAMtoGFF (.bam) = s_merge_markdup.mkdupbam
BAMtoGFF (.bai) = s_merge_mkdup.indexbam

Reply:
Ah, I see that `indexstats` performs multiple functions, neither of which are `samtools idxstats`. I didn't initially look at the task definition and assumed it corresponded to the samtools subcommand.